client/backoff/backoff.go

```diff
@@ -67,9 +70,9 @@ func (b *Backoff) Retry(ctx context.Context, f func() error) error {
 		return ctx.Err()
 	}
 
-	// Try calling f until it doesn't return an error or ctx is done.
+	// Try calling f while the error is retryable and ctx is not done.
 	for {
-		if err := f(); err != nil {
+		if err := f(); IsRetryable(err) {
```

Comment:
A reasonable next step would be allowing the client to specify a custom "is retriable" predicate, e.g. for non-gRPC.
Maybe TODO?

Reply:
My personal thought is that if clients should implement the retry loop themselves and call `b.Duration()` directly in order to handle any complex retry logic.

Reply:
SGTM. Also, they can do smth like [this](https://github.com/google/certificate-transparency-go/blob/04d4fbb06517a10291ff56ddbc11aa7390b8393a/trillian/migrillian/core/trillian.go#L123).